src/ts/common/ui/modal.ts

```diff
@@ -93,31 +97,60 @@ const createTextContainer = (text: string, subTexts?: string[]) => {
 	return container;
 };
 
-const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Promise<void>) => {
+const createOnClick = (exit: () => void, onClick?: () => Promise<void>) => {
 	const callback = onClick ?? (() => Promise.resolve());
-	element.addEventListener("click", () => callback().finally(exit));
+	return () => callback().finally(exit);
 };
 
-const dismissModals = (): void => activeOverlays.forEach((modal) => modal.dispatchEvent(new Event("click")));
+const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Promise<void>) =>
+	element.addEventListener("click", createOnClick(exit, onClick));
 
-const createModal = ({text, subText, elements, onCancel, colour}: ModalOptions): void => {
-	const exit = () => {
-		activeOverlays.delete(overlay);
-		overlay.remove();
-	};
+const dismissModals = (id = DEFAULT_ID): void =>
+	activeModals.forEach((onCancel, modal) => modal.getAttribute(MODAL_ID_ATTR) === id && onCancel());
 
-	const overlay = createOverlay();
+const createDismissal = (overlay: HTMLElement) => () => {
+	activeModals.delete(overlay);
+	overlay.remove();
+};
+
+const decorateOverlay = (
+	overlay: HTMLElement,
+	id: string,
+	exitable: boolean,
+	dismiss: () => void,
+	onCancel?: () => Promise<void>
+) => {
+	const onExit = createOnClick(dismiss, onCancel);
 	overlay.classList.add("modal");
-	addOnClick(overlay, exit, onCancel);
-	activeOverlays.add(overlay);
+	overlay.setAttribute(MODAL_ID_ATTR, id);
+	if (exitable) {
+		overlay.classList.add("exitable");
+		overlay.addEventListener("click", onExit);
+	}
+	activeModals.set(overlay, onExit);
```

Comment:
Even if it's not `exitable`, you can programmatically invoke the `onCancel` provided by the user via `dismissModals`